utils/filtered_dict.py

```diff
@@ -0,0 +1,24 @@
+import re
+
+
+class FilteredDict(dict):
+
+    def __init__(self, dic, **kwargs):
+        super().__init__(**kwargs)
```

Comment:
after initiating the dict you can accept the current dict as the default, which also means that the dic argument is redundant
```suggestion
        super().__init__(**kwargs)
        self.words_dict = self.copy()
```